doc/proposals/rlp-target-gateway-resource.md

```diff
@@ -104,35 +146,76 @@ Proposed new design:
-  - rules:
-    - paths: ["/admin/toy"]
-      hosts: ["cars.toystore.com"]
-      methods: ["POST", "DELETE"]
-    configurations:
-    - actions:
-      - generic_key:
-          descriptor_value: yes
-          descriptor_key: admin
+failureModeDeny: true
+rate_limit_policies:
+  - name: toystore
+    rate_limit_domain: toystore-app
+    upstream_cluster: rate-limit-cluster
+    hostnames: ["*.toystore.com"]
+    gateway_actions:
+      - rules:
+          - paths: ["/admin/toy"]
+            methods: ["GET"]
+            hosts: ["pets.toystore.com"]
+        configurations:
+          - actions:
+            - generic_key:
+                descriptor_key: admin
+                descriptor_value: "1"
 ```
 
+Update highlights:
+* [*minor*] `rate_limit_policies` is a list instead of a map indexed by the name/namespace.
+* [*major*] no distinction between "rules" and global actions
+* [*major*] more aligned with RLS: [multiple descriptors](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/route/v3/route_components.proto#envoy-v3-api-field-config-route-v3-virtualhost-rate-limits) structured by "rate limit configurations" with matching rules
+
 ## WASM-SHIM
 
-Each rate limit policy host has a logical name as well as a set of hostnames to be applied it based on the incoming request’s host header.
+Each rate limit policy has a logical name and a set of hostnames to activate it based on the incoming request’s host header.
+
+The WASM-SHIM builds a tree based data structure holding the rate limit policies.
+The longest (sub)domain match is used to select the policy to be applied.
+Only one policy is being applied per invocation.
 
-The WASM-SHIM builds a tree based data structure holding the rate limit policies. The longest hostname match is used to select the policy to be applied. Only one policy is being applied per invocation.
+The plugin object contains a list of configurations to build a list of Envoy's RLS descriptors.
+These policy configurations are defined at
 
-The policy contains rules and configuration to build a list of Envoy's RLS descriptors. 
+```
+rate_limit_policies[*].gatewat_actions[*].configurations
+```
 
-A given configuration will be applied, and therefore the actions activated, when rule match occurs. A rule match occurs when at least one rule matches the request. If not set, it is equivalent to matching all the requests.
+For example:
+
+```yaml
+configurations:
+- actions:
+   - generic_key:
+        descriptor_key: admin
+        descriptor_value: "1"
+```
+
+Each configuration produces, at most, one descriptor. Depending on the incoming request, one configuration may or may not produce a rate limit descriptor.
+
+Each policy configuration has associated, optionally, a set of rules to match. Rules allow matching `hosts` and/or `methods` and/or `paths`. Matching occurs when at least one rule applies against the incoming request. If rules are not set, it is equivalent to matching all the requests.
```

Comment:
> If rules are not set, it is equivalent to matching all the requests.

A route-level policy with no rules should still be under the hostnames provided in the targeted `HTTPRoute` and for gateway-level policy, we need to use `*` as the hostname.

Reply:
at the wasm filter level where are no route level and gateway level policies. There are rate limit configurations. With hostnames to scope them. 

`rules` provides flexibility to decide if one "configuration" (`rate_limit_policies[*].gatewat_actions[*].configurations`) needs to be executed or not. But always for a given hostname. `hostnames` is like a configuration rule, but applied globally for all configurations at wasm filter level.

Rules allow to decide not only based on the `hosts`, but also `method` and `path`.

For example.

There is  a rate limit configuration for `*.toystore.com` and I want some rate limit descriptors configuration only for `api.toystore.com` and another set of descriptors for `admin.toystore.com`. The wasm filter config would look like this:

```yaml
failure_mode_deny: true
rate_limit_policies:
  - name: toystore
    rate_limit_domain: toystore-app
    upstream_cluster: rate-limit-cluster
    hostnames: ["*.toystore.com"]
    gateway_actions:
      - configurations:  # no rules. Applies to all *.toystore.com traffic
          - actions:
              - generic_key:
                  descriptor_key: toystore-app
                  descriptor_value: "1"
      - rules:
          - hosts: ["api.toystore.com"]
        configurations:
          - actions:
              - generic_key:
                  descriptor_key: api
                  descriptor_value: "1"
      - rules:
          - hosts: ["admin.toystore.com"]
        configurations:
          - actions:
              - generic_key:
                  descriptor_key: admin
                  descriptor_value: "1"
```

Reply:
got it. I now see how it will work out :)